Marlin/src/HAL/HAL_STM32F1/HAL_spi_Stm32f1.cpp

```diff
@@ -79,8 +79,8 @@ void spiBegin() {
   #if !PIN_EXISTS(SS)
     #error "SS_PIN not defined!"
   #endif
-  SET_OUTPUT(SS_PIN);
   WRITE(SS_PIN, HIGH);
+  SET_OUTPUT(SS_PIN);
```

Comment:
Why write to the pin before it's sure to be an output?

Reply:
I don't know how this is working on a F1. 
On a 8bit AVR, with a externally pulled up pin what was an input, this sequence would mean.
Set the internal pull up additionally - what makes no difference.
Then set the pin to be an output - what drives it high immediately.

The other sequence can have a short moment where the output is low.

Reply:
If there is no external pull up. Switching on the internal pull up will drive the pin high with low power, Switching it to an output then activates full power.

Reply:
Interesting. That's the F1 behavior? And that what is wanted? Is that better for the attached peripheral?

Reply:
In an F1 the ports are input floating by default. If the mode is changed to putput, they will be out push pull, default low.
With the previous sequence, as @AnHardt pointed out there could be a blip of the port being pulled low.
By applying it in this sequence, when the port is set to output it will correctly go straigh from input floating to output high level.